keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,730 @@
+
+This KEP will eventually also aim to have the CRI implementation populate the container level stats needed for the `/metrics/cadvisor` endpoint (or deprecate that endpoint).
+
+### Current State of Summary API
+
+Summary API has two interfaces:
+* [cAdvisor stats provider](https://github.com/kubernetes/kubernetes/blob/release-1.20/pkg/kubelet/stats/cadvisor_stats_provider.go)
+    * Calls cAdvisor directly to obtain node, pod, and container stats
+* [CRI stats provider](https://github.com/kubernetes/kubernetes/blob/release-1.20/pkg/kubelet/stats/cri_stats_provider.go#L54)
+    * Calls CRI implementation to provide some minimum container level stats, but still relies on cAdvisor for the majority of container level stats + pod level stats + node level stats
+
+#### Current Fulfiller of Summary API Stats & Future Proposal 
+
+Below is a table describing which stats come from what source now, as well a proposal of which should come from where in the future. It also includes which fields roughly correspond to fields in the `/metrics/cadvisor` endpoint, some of which will not come from the CRI for the first iteration of this KEP. See more below.
+
+|Top level object              |Field               |Level Needed     |Currently provided by:|Proposed to be provided by:|`/metrics/cadvisor`                             |
+|------------------------------|--------------------|-----------------|----------------------|---------------------------|------------------------------------------------|
+|InterfaceStats (Network)      |RxBytes             |Pod              |cAdvisor              |CRI                        |container_network_receive_bytes_total           |
+|                              |RxErrors            |Pod              |cAdvisor              |CRI                        |container_network_receive_errors_total          |
+|                              |TxBytes             |Pod              |cAdvisor              |CRI                        |container_network_transmit_bytes_total          |
+|                              |TxErrors            |Pod              |cAdvisor              |CRI                        |container_network_transmit_errors_total         |
+|                              |                    |                 |                      |                           |container_network_receive_packets_dropped_total |
+|                              |                    |                 |                      |                           |container_network_receive_packets_total         |
+|                              |                    |                 |                      |                           |container_network_transmit_packets_dropped_total|
+|                              |                    |                 |                      |                           |container_network_transmit_packets_total        |
```

Comment:
What's the status of these empty stats? Can you update the table to reflect what we want to do with them?

Reply:
I'll update the table. Al the unlabelled fields are provided by cAdvisor, and would need to be provided by CRI or dropped